### PR TITLE
[stdlib] Constraining BinaryInteger.Words to Sequence at least

### PR DESCRIPTION
**Pull request by moiseev**

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please test and merge

**Comment by moiseev** [comment]
@swift-ci Please smoke test macOS platform

File: stdlib/public/core/Integers.swift.gyb [hunk 1]
```diff
@@ -1479,7 +1479,7 @@ public protocol BinaryInteger :
   // FIXME(integers): add doc comments
   // FIXME: Should be `Words : Collection where Words.Iterator.Element == UInt`
   // See <rdar://problem/31798916> for why it isn't.
-  associatedtype Words
+  associatedtype Words : Sequence
```
Comment (dabrahams) [inline stdlib/public/core/Integers.swift.gyb]: Can you also add the most-derived `_Indexable` protocol you can get away with here?

Comment (moiseev) [inline stdlib/public/core/Integers.swift.gyb]: FYIL This constraint caused troubles [elsewhere](https://github.com/apple/swift/pull/9047).

**Pull request merged**